lib/KBox/src/KMessage.h

```diff
@@ -191,6 +196,94 @@ class IMUMessage: public KMessage {
     };
 };
 
+class NAVMessage: public KMessage {
+  private:
+    bool apMode, apHeadingMode, apWaypointMode, apDodgeMode;
+    double currentHeading, targetHeading, courseToWaypoint;
```

Comment:
I can see your point on having the AutoPilotTask getting the current heading directly from the IMU message.  I'll tackle that.

We can take out courseToWaypoint if you like.  It was there as a placeholder for the next stage in the AP which would be adding the ability to navigate to a waypoint provided from a plotter.  Which raises a whole other conversation ... probably should have it elsewhere, so I'll put it on the discussion list ... which is can we get a message back over WiFI from a mobile device running nav software ?

With respect to replacing the mode variables with an ENUM, I thought about that when you suggested it before and started to do it, but then I realized there were issues with that.  Let's say we have 4 ENUMs that are mutually exclusive: OFF, HEADINGMODE, DODGEMODE, WAYPOINTMODE. Let's say we're in WAYPOINTMODE and we go into DODGEMODE to avoid a windsurfer.  When we come out of DODGEMODE, how does KBOX know what mode to go into then?  Does it go back to HEADINGMODE or WAYPOINTMODE?  We wouldn't know unless we track another various like previousMode. Also, it means that when we go into DODGEMODE, KBOX can keep the nav calculations going for either HEADINGMODE or WAYPOINTMODE in the background and resume giving appropriate steering directions when dropping out of DODGEMODE. Given these scenarios, having thought about it, I decided it would be cleaner to have independent boolean variables than going with your ENUM suggestion.  I'm happy with that decision, but would be interested if you think there is a cleaner solution now that you understand more about why I made the choice I did.